src/org/mozilla/javascript/NativeString.java

```diff
@@ -941,6 +946,47 @@ protected int findPrototypeId(Symbol k)
         return 0;
     }
 
+    /**
+     * <h1>String.raw (template, ...substitutions)</h1>
+     * <p>22.1.2.4 String.raw [Draft ECMA-262 / April 28, 2021]</p>
+     */
+    private static CharSequence js_raw(Context cx, Scriptable scope, Object[] args) {
+        final Object undefined = Undefined.instance;
+        /* step 1-2 */
+        Object arg0 = args.length > 0 ? args[0] : undefined;
+        Scriptable cooked = ScriptRuntime.toObject(cx, scope, arg0);
+        /* step 3 */
+        Object rawValue = cooked.get("raw", cooked);
+        if (rawValue == NOT_FOUND) rawValue = undefined;
```

Comment:
FWIW if I am reading "./gradlew jacocoTestReport" correctly, the "NOT_FOUND" case here isn't actually being tested.

Reply:
Updated this to use `ScriptRuntime.getObjectProp` instead, which does the correct ecma `[[Get]]` operation. It now searches up the prototype chain and returns undefined if not found in a single call. The test that covers raw being undefined is `template-raw-not-object-throws.js`